BloomFilter.md

```diff
@@ -147,21 +147,30 @@ This closes the definition of a block and the operations on it.
 
 Now that a block is defined, we can describe Parquet's split block
 Bloom filters. A split block Bloom filter (henceforth "SBBF") is
-composed of `z` blocks, where `z` is a power of two greater than or
-equal to one and less than 2 to the 27th power. When an SBBF is
-initialized, each block in it is initialized, which means each bit in
-each word in each block in the SBBF is unset.
+composed of `z` blocks, where `z` is greater than or equal to one and
+less than 2 to the 31st power. When an SBBF is initialized, each block
+in it is initialized, which means each bit in each word in each block
+in the SBBF is unset.
 
 In addition to initialization, an SBBF supports an operation called
 `filter_insert` and one called `filter_check`. Each takes as an
 argument a 64-bit unsigned integer; `filter_check` returns a boolean
 and `filter_insert` does not return a value, but does modify the SBBF.
 
 The `filter_insert` operation first uses the most significant 32 bits
-of its argument, modulo the number of blocks, to select a block to
-operate on. It then uses the least significant 32 bits of the argument
-to `filter_insert` as an argument to `block_insert` called on that
-block.
+of its argument to select a block to operate on. If the number of
+blocks is `z`, the most significant 32 bits of the `filter_insert`
+argument are multiplied by `z`; the most significant 32 bits of this
+product are the index of the block to operate on. The `filter_insert`
+function then uses the least significant 32 bits of the argument to
+`filter_insert` as an argument to `block_insert` called on that block.
+
+The technique for converting the most significant 32 bits to an
+integer between 0 and z-1 (inclusive) avoids using the modulo
+operation, which is often very slow.  The oldest reference I know to
```

Comment:
Nit: first person statements in a spec seem strange. Can we rephrase this to something like "this trick was found via Kennith A. Ross's report ..."?

Reply:
Done.